src/main.cpp

```diff
@@ -2906,33 +2908,63 @@ TbBool setup_move_out_of_cave_in(struct Thing *thing)
     {
         struct CreatureControl* cctrl;
         cctrl = creature_control_get_from_thing(thing);
-        if ( setup_person_move_to_coord(thing, &cctrl->flee_pos, 0) )
+        long dist = LbDiagonalLength(abs(thing->mappos.x.val - cctrl->flee_pos.x.val), abs(thing->mappos.y.val - cctrl->flee_pos.y.val));
+        // If you're too close to the flee position, no point in going there to escape cave in damage.
+        if (dist <= 200)
```

Comment:
What is 200? Could you please `#define CAVE_IN_RETREAT_DIST 200` at start of a file and use it instead of 200

Reply:
ok